langroid/agent/special/sql_chat_agent.py

```diff
@@ -0,0 +1,235 @@
+"""
+Agent that allows interaction with an SQL database using SQLAlchemy library. 
+The agent can execute SQL queries in the database and return the result. 
+
+Functionality includes:
+- adding table and column context
+- asking a question about a SQL schema
+"""
+import logging
+from typing import Dict, Optional, Union
+
+from prettytable import PrettyTable
+from rich.console import Console
+from sqlalchemy import MetaData, create_engine, inspect, text
+from sqlalchemy.orm import sessionmaker
+
+from langroid.agent.chat_agent import ChatAgent, ChatAgentConfig
+from langroid.agent.tool_message import ToolMessage
+from langroid.language_models.openai_gpt import OpenAIChatModel, OpenAIGPTConfig
+from langroid.prompts.prompts_config import PromptsConfig
+from langroid.vector_store.base import VectorStoreConfig
+
+logger = logging.getLogger(__name__)
+
+console = Console()
+
+DEFAULT_TABLE_CHAT_SYSTEM_MESSAGE = """
+You are a savvy data scientist, with expertise in analyzing SQL database,
+using Python and the SQLAlchemy library for data manipulation.
+You do not have access to the database 'db' directly, so you will need to use the 
+`run_code` tool/function-call to answer the question.
+The tables in the database are:
```

Comment:
change to: "Here is information about the tables, columns and their relationships:..."